src/assets/Generator.Shared/OperationInternals.cs

```diff
@@ -165,8 +165,8 @@ public T Value
             }
         }
 
-        public bool HasCompleted => _hasCompleted;
-        public bool HasValue => _hasValue;
+        public virtual bool HasCompleted => _hasCompleted;
```

Comment:
Do these need to be virtual given that I don't see anyone inheriting this class?

Reply:
That is a great question, given that is just copy pasted from the old implementation I prefer if I don't mess with that in this PR.

If you are confident I can quickly remove the virtual here let me know

Reply:
Seems safe if we decided to go with composition in `OperationOrResponseInternals`